swift/ql/lib/codeql/swift/elements/expr/ApplyExpr.qll

```diff
@@ -1,9 +1,25 @@
 private import codeql.swift.generated.expr.ApplyExpr
 private import codeql.swift.elements.decl.AbstractFunctionDecl
 private import codeql.swift.elements.expr.DeclRefExpr
+private import codeql.swift.elements.expr.MethodRefExpr
+private import codeql.swift.elements.expr.ConstructorRefCallExpr
 
 class ApplyExpr extends ApplyExprBase {
-  AbstractFunctionDecl getStaticTarget() { result = this.getFunction().(DeclRefExpr).getDecl() }
+  AbstractFunctionDecl getStaticTarget() {
+    exists(Expr f |
+      f = this.getFunction() and
+      (
+        result = f.(DeclRefExpr).getDecl()
+        or
+        result = f.(MethodRefExpr).getMethod()
+        or
+        result = f.(ConstructorRefCallExpr).getFunction().(DeclRefExpr).getDecl()
```

Comment:
I imagine we can remove this case once `MethodRefExpr` includes `ConstructorRefCallExpr`.

Reply:
it depends how we want to do that. If we want constructors to be in the same bag as methods, then yes, though it will mean having to make `MethodRefExpr` a superclass of two distinct concrete IPA classes (one synthesized from `DotSyntaxCallExpr` and the other from `ConstructorRefCallExpr`). The other approach would be to make `ConstructorRefExpr` a sibling of `MethodRefExpr` under `LookupExpr`, then we would edit this line rather than suppress it

Reply:
Good point. In this case, I think we should follow whatever precedence Java/C# has set. Do you know if their version of `MethodRef` also includes constructors (assuming you can actually do this in those languages 🤔)?